cloudinit/sources/DataSourceAzure.py

```diff
@@ -303,14 +305,15 @@ def __init__(self, sys_cfg, distro, paths):
         self.dhclient_lease_file = self.ds_cfg.get("dhclient_lease_file")
         self._network_config = None
         self._ephemeral_dhcp_ctx = None
+        self._wireserver_endpoint = DEFAULT_WIRESERVER_ENDPOINT
         self.iso_dev = None
 
     def _unpickle(self, ci_pkl_version: int) -> None:
         super()._unpickle(ci_pkl_version)
 
         self._ephemeral_dhcp_ctx = None
-        if not hasattr(self, "iso_dev"):
-            self.iso_dev = None
+        self.iso_dev = None
```

Comment:
This means the iso won't be ejected in init-network for normal (non-preprovisioned) VM.